crates/diesel_models/src/query/customers.rs

```diff
@@ -73,6 +73,21 @@ impl Customer {
         .await
     }
 
+    #[instrument(skip(conn))]
+    pub async fn list_by_merchant_id(
+        conn: &PgPooledConn,
+        merchant_id: &str,
+    ) -> StorageResult<Vec<Self>> {
+        generics::generic_filter::<<Self as HasTable>::Table, _, _, _>(
+            conn,
+            dsl::merchant_id.eq(merchant_id.to_owned()),
+            None,
```

Comment:
Should we add some limit here?

Reply:
I will add limits and offsets in a separate PR, I want to add pagination using the `primary_keys` and not limits and offsets as this will be a costly operation